cpp/src/arrow/compute/exec/key_encode.cc

```diff
@@ -1366,8 +1377,8 @@ void KeyEncoder::KeyRowMetadata::FromColumnMetadataVector(
   // a) Boolean column, marked with fixed-length 0, is considered to have fixed-length
   // part of 1 byte. b) Columns with fixed-length part being power of 2 or multiple of row
   // alignment precede other columns. They are sorted among themselves based on size of
-  // fixed-length part. c) Fixed-length columns precede varying-length columns when both
-  // have the same size fixed-length part.
+  // fixed-length part decreasing. c) Fixed-length columns precede varying-length columns
```

Comment:
"They are sorted in decreasing order of the size of their fixed-length part"

Reply:
changed